Unable to add member after removing it without deleting data-dir first

Start a three-node cluster:

```
% ./bin/etcd -name=node1 -initial-cluster=node1=http://127.0.0.1:7001,node2=http://127.0.0.1:7002,node3=http://127.0.0.1:7003 -listen-peer-urls=http://127.0.0.1:7001 -initial-advertise-peer-urls=http://127.0.0.1:7001 -listen-client-urls=http://127.0.0.1:4001 -advertise-client-urls=http://127.0.0.1:4001

% ./bin/etcd -name=node2 -initial-cluster=node1=http://127.0.0.1:7001,node2=http://127.0.0.1:7002,node3=http://127.0.0.1:7003 -listen-peer-urls=http://127.0.0.1:7002 -initial-advertise-peer-urls=http://127.0.0.1:7002 -listen-client-urls=http://127.0.0.1:4002 -advertise-client-urls=http://127.0.0.1:4002

% ./bin/etcd -name=node3 -initial-cluster=node1=http://127.0.0.1:7001,node2=http://127.0.0.1:7002,node3=http://127.0.0.1:7003 -listen-peer-urls=http://127.0.0.1:7003 -initial-advertise-peer-urls=http://127.0.0.1:7003 -listen-client-urls=http://127.0.0.1:4003 -advertise-client-urls=http://127.0.0.1:4003
```

Wait for the cluster to stabilize, then remove node3:

```
% curl -i -X DELETE http://127.0.0.1:4001/v2/admin/members/2745e2525fce8fe
```

Note the logs on node1 and node2:

```
2014/10/29 14:15:39 etcdhttp: remove node 2745e2525fce8fe
```

node3 will panic with the following information

```
2014/10/29 14:15:40 the member has been removed
```

Now add node3 back to the cluster:

```
% curl -X POST http://127.0.0.1:4002/v2/admin/members -H "Content-Type: application/json" -d '{"peerURLs":["http://127.0.0.1:7003"]}'
{"id":"9690ffae174d6fd9","peerURLs":["http://127.0.0.1:7003"]}
```

And finally, try to start node3:

```
% ./bin/etcd -name=node3 -initial-cluster=node1=http://127.0.0.1:7001,node2=http://127.0.0.1:7002,node3=http://127.0.0.1:7003 -listen-peer-urls=http://127.0.0.1:7003 -initial-advertise-peer-urls=http://127.0.0.1:7003 -listen-client-urls=http://127.0.0.1:4003 -advertise-client-urls=http://127.0.0.1:4003 -initial-cluster-state=existing
2014/10/29 14:16:31 etcd: no data-dir provided, using default data-dir ./node3.etcd
2014/10/29 14:16:31 etcdserver: restart member 2745e2525fce8fe in cluster 1f4875d9dbf3e329 at commit index 73
2014/10/29 14:16:31 etcd: listening for peers on http://127.0.0.1:7003
2014/10/29 14:16:31 etcd: listening for client requests on http://127.0.0.1:4003
2014/10/29 14:16:33 the member has been removed
panic: the member has been removed
```

node1 and node2 both log this:

```
2014/10/29 14:16:33 etcdhttp: error processing raft message: etcdserver: server removed
```

If I remove the data-dir, I can successful start node3, but at no point was this obvious to me as the operator. Maybe etcd should fail to start if it realizes it has a different ID in the local data-dir than the ID it's trying to assume in the cluster?

--- a/etcdserver/sender.go
+++ b/etcdserver/sender.go
@@ -109,11 +109,12 @@ func httpPost(c *http.Client, url string, cid types.ID, data []byte) bool {
 	switch resp.StatusCode {
 	case http.StatusPreconditionFailed:
 		// TODO: shutdown the etcdserver gracefully?
-		log.Fatalf("etcd: conflicting cluster ID with the target cluster (%s != %s). Exiting.", resp.Header.Get("X-Etcd-Cluster-ID"), cid.String())
+		log.Fatalf("etcd: conflicting cluster ID with the target cluster (%s != %s)", resp.Header.Get("X-Etcd-Cluster-ID"), cid.String())
 		return false
 	case http.StatusForbidden:
 		// TODO: stop the server
-		log.Fatalf("etcd: this member has been permanently removed from the cluster. Exiting.")
+		log.Println("etcd: this member has been permanently removed from the cluster")
+		log.Fatalln("etcd: the data-dir used by this member must be removed so that this host can be re-added with a new member ID")
 		return false
 	case http.StatusNoContent:
 		return true

